lib/hexpm/accounts/session.ex

```diff
@@ -1,25 +1,62 @@
 defmodule Hexpm.Accounts.Session do
   use Hexpm.Schema
+  alias Hexpm.Accounts.User
 
   schema "sessions" do
-    field :token, :binary
+    belongs_to :user, User
+    field :uuid, Ecto.UUID, autogenerate: true
+    field :token_hash, :binary
     field :data, :map
+    field :expires_at, :utc_datetime_usec
+    field :active, :boolean
+
     timestamps()
   end
 
-  def build(data) do
-    change(%Session{}, data: data, token: :crypto.strong_rand_bytes(96))
+  def build(user, data, token) do
+    expires_at = DateTime.add(DateTime.utc_now(), data.expires_in, :second)
+    token_hash = :crypto.hash(:blake2b, token)
```

Comment:
Might need to adjust algo here, I see tests are failing CI because of this, but not sure what hexpm is using in prod.